### PR TITLE
Feature/atomizie use fetch

### PR DESCRIPTION
**Pull request by ricokahler**
This splits out `useFetch` into `useData` and `useStatus`

This PR also add `useError`. Error states are _not_ shared in the current implementation.

I don't know if we need to share errors but I'm open to what you guys think.

Addresses #21  and #23

### CODE REVIEW

File: src/Guard/Guard.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+import useStatus from '../useStatus';
 
-function Guard({ fetch, children }) {
-  const [data, status] = useFetch(fetch);
+function Guard({ fetch, children, shouldShowWhen = isNormal }) {
+  const data = useData(fetch);
+  const status = useStatus(fetch);
 
-  return isNormal(status) && children(data);
+  return shouldShowWhen(status) && children(data);
```
Comment (MPiccinato) [inline src/Guard/Guard.js]: Didn't we agree to leave out the `shouldShowWhen` param?

File: src/useFetch/useFetch.js [hunk 2]
```diff
@@ -1,36 +1,12 @@
...
-
-  function returnPreviousIfUnchanged(...args) {
-    const next = fn(...args);
-    if (shallowEqual(next, previous)) return previous;
-    previous = next;
-    return next;
-  }
-
-  return returnPreviousIfUnchanged;
+  return useMemo(() => [data, status], [data, status]);
```
Comment (MPiccinato) [inline src/useFetch/useFetch.js]: 👍

File: src/Guard/Guard.js (hunk 1 above)
Comment (ricokahler) [reply src/Guard/Guard.js]: I think we did. I forgot tbh. I can remove it before it hits master

Comment (ricokahler) [reply src/Guard/Guard.js]: done.

**Pull request merged**